name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/actions/InsertAction.java

```diff
@@ -51,6 +57,18 @@ public Status process(AccountTransaction transaction, Account account)
         if (transaction.getSecurity() != null)
             process(transaction.getSecurity());
         account.addTransaction(transaction);
+
+        if (removeDividends
+            && transaction.getType() == AccountTransaction.Type.DIVIDENDS && transaction.getSecurity() != null)
```

Comment:
@buchen Verstehe ich richtig, dass `AccountTransaction.Type.DIVIDENDS` in Wirklichkeit ein Bool ist und ich deshalb `transaction.getSecurity() != null` brauche?

Reply:
@Al2Klimov Die AccountTransaction hat eine Referenz auf das Wertpapier (security) die bei Dividenden gesetzt sein muss. Bei anderen Buchungen (wie Einlage, Entnahme) nicht oder optional (Steuern, Gebühren). Der "type" ist kein Boolean sondern eine Enum mit den verschiedenen Buchungstypen.

An dieser Stelle müsste meiner Meinung nach die Prüfung auf type == DIVIDENDS reichen. An anderer Stelle würde geprüft, dass Dividenden auch ein Wertpapier haben.

Reply:
> kein Boolean sondern eine Enum

Aber die haben alle Bools in Klammern hintendran. Was hat das zu bedeuten?